common/src/main/scala/react/resizeDetector/hooks.scala

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.resizeDetector
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.{ facade => Raw }
+import org.scalajs.dom.html
+import react.common._
+import react.resizeDetector.ResizeDetector._
+
+import scala.annotation.nowarn
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.scalajs.js.|
+
+@js.native
+protected trait ReactResizeDetectorDimensions extends js.Object {
+  val height: js.UndefOr[Raw.JsNumber]
+  val width: js.UndefOr[Raw.JsNumber]
+}
+
+@js.native
+trait UseResizeDetectorReturnJS extends ReactResizeDetectorDimensions {
+  val ref: Raw.React.RefFn[html.Element]
+}
+
+sealed trait UseResizeDetectorReturn {
+  val height: Option[Int]
+  val width: Option[Int]
+  val ref: Ref.Simple[html.Element]
+}
+
+object UseResizeDetectorReturn {
+  def fromJS(r: UseResizeDetectorReturnJS): UseResizeDetectorReturn = new UseResizeDetectorReturn {
+    val height = r.height.toOption.map(_.toInt)
+    val width  = r.width.toOption.map(_.toInt)
+    val ref    =
+      Ref.fromJs(r.ref.asInstanceOf[facade.React.RefHandle[html.Element | Null]])
+  }
+
+  implicit val reusabilityUseResizeDetectorReturn: Reusability[UseResizeDetectorReturn] =
+    Reusability.by(x => (x.height, x.width))
+}
```

Comment:
Can we extract the traits defined in `ResizeDetector` and reuse them in both places?

Reply:
They are not exactly the same. Instead of `ref` it has a `targetRef`. I was hoping to delete the component based one later

Reply:
I missed that. We can still reuse `DimensionsJS`. We can still remove the class component version later. `Props` is being reused.